pkg/node/bootstrap.go

```diff
@@ -127,13 +126,8 @@ func bootstrapNode(
 	}
 	b.hiveCloser = hive
 
-	metricsDB, err := shed.NewDBWrap(stateStore.DB())
-	if err != nil {
-		return nil, fmt.Errorf("unable to create metrics storage for kademlia: %w", err)
-	}
-
-	kad, err := kademlia.New(swarmAddress, addressbook, hive, p2ps, &noopPinger{}, metricsDB, logger,
-		kademlia.Options{Bootnodes: bootnodes, BootnodeMode: o.BootnodeMode, StaticNodes: o.StaticNodes})
+	kad, err := kademlia.New(swarmAddress, addressbook, hive, p2ps, &noopPinger{}, stateStore, logger,
+		kademlia.Options{Bootnodes: bootnodes, BootnodeMode: o.BootnodeMode, StaticNodes: o.StaticNodes, DataDir: o.DataDir})
```

Comment:
For the bootstrapper it maybe okay to have an in mem metrics repo. It is fine either way.

Reply:
Before it was persisted so I'd suggested keeping it that way.